agent/agent.go

```diff
@@ -785,6 +796,9 @@ func (a *Agent) runOutputs(
 	cancel()
 	wg.Wait()
 
+	log.Printf("I [agent] Stopping running outputs")
+	stopRunningOutputs(unit.outputs)
```

Comment:
Can't you close the outputs in `a.flushLoop()` in the loop above?

Reply:
I'd rather close it here.